### ISSUE TITLE
DatabaseError: SSL SYSCALL error: EOF detected

### ISSUE THREAD

**Issue by rugging24** [psycopg/psycopg2#533]
Hello,
I currently adopted psycopg2 for logical decoding. It runs very fine for a few hours and then it throws the error : 
DatabaseError: SSL SYSCALL error: EOF detected
just at the point : consume_stream(consume)

and doesn't reconnect until a few hours later.

any ideas ?

running PG9.5 and psycopg2.7.1

thanks

**Comment by dvarrazzo** [comment]
Psycopg is just reporting you are being disconnected as far as I can see. I think you should ask around to the postgres mailing lists (e.g. pgsql-general) to understand the cause of disconnection. 

Closing on this side: if you discover any error the adapter is making and which may affect the connection let us know and we will try to fix it.

**Comment by gauravssingh** [comment]
Your query might too large - too many operations. This can cause the system to run out of memory.

**Comment by bagrym** [comment]
Hi rugging24! Did you figure out the root cause? My app is also running fine for few days, after some idle period I get error. After restart all goes well... for a while. Thanks

**Comment by qris** [comment]
I saw this in the logs on our server, could be related:

> 2018-10-03 15:19:21.186 GMT [18023]: [11] user=...,db=...,app=[unknown],host=... LOG:  terminating walsender process due to replication timeout

This implies that the client might be taking too long to process a large message. From [this DBA.se post](https://dba.stackexchange.com/a/137936/50610), it looks like you can increase the `wal_sender_timeout`:

> The server error message means that the client did not send a status update within `wal_sender_timeout` milliseconds, [see documentation](http://www.postgresql.org/docs/current/static/runtime-config-replication.html#GUC-WAL-SENDER-TIMEOUT).
>
> The basebackup needs to complete before this `wal_sender_timeout` period, else the server resets the connection.

**Comment by stevenchew83** [comment]
> Your query might too large - too many operations. This can cause the system to run out of memory.

agree.

**Comment by tristann3** [comment]
I have searched SO and have found myself here, I can see a lot of solutions including the queries being too large but I am getting this error when I log into my site.

I am currently hosting a Flask, Postgres website on Heroku Hobby plan. When I log in I get a stack trace ending in 
`sqlalchemy.exc.OperationalError: (psycopg2.OperationalError) SSL SYSCALL error: EOF detected`

This is probably my 2nd or 3rd time hosting something on Heroku. I made a change to my flask models and had to update my database on Heroku to reflect the changes, I went down a rabbit hole and eventually came across something in Heroku called pg:copy. I ran into some issues following instructions but when I (think) the problem was solved, this psycopg2 error arose. 

TLDR^ I'm not sure if this is related to the issue I am having.

**Comment by qris** [comment]
There are multiple possible causes for this problem, and they all lie on the server, which disconnected the client without giving a reason. You are seeing the client's complaint about being disconnected, but that doesn't tell you why. The answer is probably in the server logs. This is not a psycopg2 bug.
